jetstream/statistics.py

```
@@ -511,44 +511,51 @@ def transform(
-                        statistic="kernel_density_estimate",
-                        parameter=x,
-                        branch=branch,
-                        comparison=None,
-                        comparison_to_branch=None,
-                        ci_width=None,
-                        point=y,
-                        lower=None,
-                        upper=None,
+                result = kde.evaluate(grid.grid)
+                if group[metric].min() == 0 and grid.geometric:
+                    results.append(
+                        StatisticResult(
+                            metric=metric,
+                            statistic="kernel_density_estimate",
+                            parameter=0,
+                            branch=branch,
+                            comparison=None,
+                            comparison_to_branch=None,
+                            ci_width=None,
+                            point=kde.evaluate(0)[0],
+                            lower=None,
+                            upper=None,
+                        )
+                    )
+                for x, y in zip(grid.grid, result):
+                    results.append(
+                        StatisticResult(
+                            metric=metric,
+                            statistic="kernel_density_estimate",
+                            parameter=x,
+                            branch=branch,
+                            comparison=None,
+                            comparison_to_branch=None,
+                            ci_width=None,
+                            point=y,
+                            lower=None,
+                            upper=None,
+                        )
                     )
+            except Exception as e:
+                logger.error(
+                    f"KernelDensityEstimate error for metric {metric}, branch {branch}: {e}",
+                    extra={"experiment": experiment.normandy_slug},
```

Not sure if we want to raise the exception and let it fail or just log the error and keep going with the analysis

---

Can we apply this at a point where it would catch all statistic invocations instead of making it KDE-specific?

I think generating a good error message but allowing the analysis to complete makes the most sense to me; this doesn't necessarily indicate that anything else is wrong with the experiment.